submodules-tasks.yml

```
@@ -0,0 +1,364 @@
+      DESTINATION: "{{.G_CORE_SUBMODULE_DIR}}/sqlite3"
+    sources:
+      - "{{.G_DEP_DOWNLOAD_SCRIPT}}"
+      - "{{.TASKFILE}}"
+    generates: [ "{{.CHECKSUM_FILE}}" ]
+    deps:
+      - ":init"
+      - task: ":utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.DESTINATION}}"
+    cmds:
+      - task: "download-dependency"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          SOURCE_URL: "https://www.sqlite.org/2021/sqlite-amalgamation-3360000.zip"
+          SOURCE_NAME: "sqlite-amalgamation-3360000"
+          FLAGS: "--extract --no-submodule"
+          DESTINATION: "{{.DESTINATION}}"
+      # This command must be last
+      - task: ":utils:compute-checksum"
+        vars:
+          DATA_DIR: "{{.DESTINATION}}"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+  yaml-cpp:
+    internal: true
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK | replace \":\" \"#\"}}.md5"
+      DESTINATION: "{{.G_CORE_SUBMODULE_DIR}}/yaml-cpp"
+    sources:
+      - "{{.G_DEP_DOWNLOAD_SCRIPT}}"
+      - "{{.TASKFILE}}"
+    generates: ["{{.CHECKSUM_FILE}}"]
+    deps:
+      - ":init"
+      - task: ":utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.DESTINATION}}"
+    cmds:
+      - task: "download-dependency"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          SOURCE_URL: "https://github.com/jbeder/yaml-cpp/archive/refs/tags/yaml-cpp-0.7.0.zip"
+          SOURCE_NAME: "yaml-cpp-yaml-cpp-0.7.0"
+          FLAGS: "--extract"
+          DESTINATION: "{{.DESTINATION}}"
+      # This command must be last
+      - task: ":utils:compute-checksum"
+        vars:
+          DATA_DIR: "{{.DESTINATION}}"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+  yscope-log-viewer:
+    internal: true
+    vars:
+      CHECKSUM_FILE: "{{.G_BUILD_DIR}}/{{.TASK | replace \":\" \"#\"}}.md5"
+      DESTINATION: "{{.G_LOG_VIEWER_WEBUI_SRC_DIR}}/yscope-log-viewer"
+    sources:
+      - "{{.G_DEP_DOWNLOAD_SCRIPT}}"
+      - "{{.TASKFILE}}"
+    generates: [ "{{.CHECKSUM_FILE}}" ]
+    deps:
+      - ":init"
+      - task: ":utils:validate-checksum"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          DATA_DIR: "{{.DESTINATION}}"
+    cmds:
+      - task: "download-dependency"
+        vars:
+          CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
+          SOURCE_URL: "https://github.com/y-scope/yscope-log-viewer/archive/df996ea.zip"
+          SOURCE_NAME: "yscope-log-viewer-df996eac000823d02f9cd0b9eb4bb732dd634ae5"
+          FLAGS: "--extract"
+          DESTINATION: "{{.DESTINATION}}"
+      # This command must be last
+      - task: ":utils:compute-checksum"
+        vars:
+          DATA_DIR: "{{.DESTINATION}}"
+          OUTPUT_FILE: "{{.CHECKSUM_FILE}}"
+
+
+  download-dependency:
+    internal: true
+    label: "{{.TASK | replace \":\" \"#\"}}-{{.SOURCE_NAME}}"
```

```suggestion
    label: "{{.TASK}}-{{.SOURCE_NAME}}"
```

We only need the replacement for filesystem paths.